core/src/dpdk/port.rs

```diff
@@ -437,26 +474,57 @@ impl<'a> PortBuilder<'a> {
             }
 
             // configures the TX queue with defaults
-            let txq_index = TxQueueIndex(idx as u16);
+            let txq = TxQueueIndex(idx as u16);
             unsafe {
                 ffi::rte_eth_tx_queue_setup(
                     self.port_id.0,
-                    txq_index.0,
+                    txq.0,
                     self.txd,
                     socket_id.0 as raw::c_uint,
                     ptr::null(),
                 )
                 .to_result()?;
             }
 
-            let queue = PortQueue {
-                port_id: self.port_id,
-                rxq_index,
-                txq_index,
-                kni: kni.as_ref().map(|v| v.txq()),
-            };
+            let mut q = PortQueue::new(self.port_id, rxq, txq);
+
+            if let Some(kni) = &kni {
+                q.set_kni(kni.txq());
+            }
+
+            #[cfg(feature = "metrics")]
+            {
+                // have space to set up the stats per core.
+                if ffi::RTE_ETHDEV_QUEUE_STAT_CNTRS >= len as u32 {
+                    unsafe {
+                        ffi::rte_eth_dev_set_rx_queue_stats_mapping(
+                            self.port_id.0,
+                            idx as u16,
+                            idx as u8,
+                        )
+                        .to_result()?;
+
+                        ffi::rte_eth_dev_set_tx_queue_stats_mapping(
+                            self.port_id.0,
+                            idx as u16,
+                            idx as u8,
+                        )
+                        .to_result()?;
+                    }
+                }
+
+                // counter to track dropped TX packets.
+                let counter = SINK.scoped("port").counter_with_labels(
+                    "dropped",
+                    labels!(
+                        "port" => self.name.clone(),
```

Comment:
are these labels just for hierarchy categorization? Should we think of any others to have dimensions on, and like do we not wanted dropped per core for instance?

Reply:
labels will let you do aggregation easier, like all rx traffic for `eth1` across the cluster etc. at least for something like prometheus.

also I actually did do tx dropped per core. but then realized DPDK doesn't do it per core for rx, so decided to keep it consist. though we do lose some information we may have otherwise.

Reply:
ah, interesting, ok on the dropped side.